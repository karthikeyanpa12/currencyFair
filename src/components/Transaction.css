.transaction-container{
    flex:2 0 200px;
    padding: 24px;
} 

.transaction{
    flex:2 0 300px;
}

.info{
    font-size:14px;
    padding: 12px 12px 24px 0px;
    color:#808080;
}

.money-send{
    border: 1px solid #749dab;
    padding: 24px;
    background-color:#f0f2f3;
}

.money-send .you-send{ 
    color:#5ccada;
}

.label{
    padding-bottom: 12px;
}

.float-right{
    float:right;
    padding-right: 12px;
}

.money-reciever{ 
    border: 1px solid #749dab;
    padding: 24px;
    margin: 24px 0px 24px 0px;
}

.restrict{
    font-size: 10px;
    color:lightblue;
    padding: 12px;
}

.receiver{
    color:#d3d3d3;
}

.sent-container{
    flex:1 0 200px !important;
    padding:24px;
}

.sent-details{
    margin:24px;
    border: 1px solid lightgrey; 
}

.sent-details .receiver{
    border-bottom:1px solid lightgrey; 
    margin: 12px 0px 12px 0px;
}

.align-left{
    padding: 12px;
    color:lightgray;
}

.sent-border{
    border: 1px solid lightgrey;
    padding: 12px; 
    margin:0px 0px 24px 0px;
    color:black;
}

.color-blue{
  color:#3a8096;
}

.saved{
    padding-left:24px;
    color: orange;
    margin-bottom: 0px;
}

button{
    background-color: #61bfde;
    width:80px;
    height:30px;
    color:white;
}

.verify-button{
    width: 120px;
}