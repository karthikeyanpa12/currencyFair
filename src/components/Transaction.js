import React, { Component } from "react";
import Modal from './Modal';
import { connect } from 'react-redux';
import { getTransaction } from '../action';
import "./Transaction.css";

export class Transaction extends Component {
    state = { showModal: false };

    componentDidMount() {
        this.props.getTransaction();
    }

    enableModal = () => this.setState({ showModal: true })
    onClose = () => this.setState({ showModal: false })

    render() {
        const { showModal } = this.state;
        const { rate, fee } = this.props.transaction;
        const receiver = (2000.00 * rate) - fee;
        return (
            <div className="transaction-container">
                <div className="transaction">
                    <span>Let's set up your transaction!</span>
                    <div className="info">Specify the amount to be sent or received</div>
                    <div className="money-send">
                        <div className="label">
                            <span className="you-send">YOU SEND</span>
                            <span className="float-right">EUR</span>
                        </div>
                        <div>£ 2,000.00 </div>
                    </div>
                    <div className="money-reciever">
                        <div className="label">
                            <span className="receiver">RECEIVER GETS</span>
                            <span className="float-right">GBP</span>
                        </div>
                        <div>£ {receiver}  </div>
                    </div>
                </div>
                <Modal show={showModal} onClose={this.onClose}></Modal>
                <button onClick={this.enableModal}> Next</button>
            </div>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        transaction: state.transaction
    };
};


export default connect(mapStateToProps, { getTransaction })(Transaction);

