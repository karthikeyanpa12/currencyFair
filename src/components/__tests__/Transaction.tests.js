import React from 'react';
import "babel-polyfill";
import { Transaction } from '../Transaction';
import { shallow } from 'enzyme';

describe('Transaction', () => {
    const transaction = { rate: 0.5, fee: 2.5, deliveryDate: '29-sep-2019' }
    const tree = shallow(<Transaction getTransaction={jest.fn()} transaction={transaction} />);
    let instance = tree.instance();

    test('renders correctly', async () => {
        expect(tree).toMatchSnapshot();
    });

    test('enableModal', async () => {
        await instance.enableModal();
        expect(instance.state.showModal).toBe(true);
    });

    test('onClose', async () => {
        await instance.onClose();
        expect(instance.state.showModal).toBe(false);
    });

});
