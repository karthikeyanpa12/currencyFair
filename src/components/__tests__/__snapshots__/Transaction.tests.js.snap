// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transaction renders correctly 1`] = `ShallowWrapper {}`;
